src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/proxy/ProxyControl.java

```diff
@@ -355,6 +399,54 @@ public void setHTTPSampleNamingMode(int httpNamingMode) {
         setProperty(new IntegerProperty(HTTP_SAMPLER_NAMING_MODE, httpNamingMode));
     }
 
+    public void setHttpSampleNumberingMode(int httpSampleNumberingMode) {
+        setProperty(new IntegerProperty(HTTP_SAMPLER_NUMBERING_MODE, httpSampleNumberingMode));
+        if (httpSampleNumberingMode == 0) {
+            // choose prefix numbering
+            JMeterUtils.setProperty("proxy.number.mode", "prefix"); // $NON-NLS-1$
```

Comment:
This way of communicating through System property is not clean, it break object isolation and might make difficult to maintain.